vassal-app/src/main/java/VASSAL/counters/BasicPiece.java

```diff
@@ -304,6 +307,23 @@ else if (CURRENT_Y.equals(key)) {
     else if (PIECE_UID.equals(key)) {
       return getId();
     }
+    else if (STACK_POS.equals(key)) {
+      final Stack parent = getParent();
+      if (parent == null) {
+        return "1";
+      }
+      final GamePiece outer = Decorator.getOutermost(this);
+      for (int i = 0; i < parent.getPieceCount(); i++) {
```

Comment:
Couldn't you use parent.indexOf(outer) instead of the loop? (Can't remember whether it's zero-based or one-based but anyway I think it could be used)

Reply:
Yes, `Statck.indexOf` returns the index of the piece or `-1` if not found, so this could be:
```
final int i = parent.indexOf(outer);
if (i != -1) {
  return String.valueOf(parent.getPieceCount() - i);
}
```